backend/protos/xyz/block/ftl/v1/schemaservice.proto

```diff
@@ -14,62 +14,97 @@ message GetSchemaResponse {
   repeated ftl.schema.v1.Changeset changesets = 2;
 }
 
-message PullSchemaRequest {}
-message PullSchemaResponse {
-  // ChangesetCreated is sent when a new changeset is created.
-  message ChangesetCreated {
-    ftl.schema.v1.Changeset changeset = 1;
-  }
+// ChangesetCreated is sent when a new changeset is created.
+message ChangesetCreatedNotification {
+  ftl.schema.v1.Changeset changeset = 1;
+}
 
-  // ChangesetFailed is sent when a changeset fails.
-  message ChangesetFailed {
-    string key = 1;
-    string error = 2;
-  }
+// ChangesetPreparedNotification is sent when a changeset is provisioned and ready to become active
+message ChangesetPreparedNotification {
+  ftl.schema.v1.Changeset changeset = 1;
+}
 
-  // ChangesetFailed is sent when a changeset becomes canonical.
-  message ChangesetCommitted {
-    string key = 1;
-  }
+// ChangesetFailed is sent when a changeset becomes canonical.
+message ChangesetCommittedNotification {
+  ftl.schema.v1.Changeset changeset = 1;
+}
+
+// ChangesetRollingBack is sent when a changeset is being rolled back without committing.
+message ChangesetRollingBackNotification {
+  string key = 1;
+  string error = 2;
+}
 
-  // Deployment created is sent when a deployment is new to the listener but is not part of a changeset.
-  message DeploymentCreated {
-    // Will not be set for builtin modules.
-    // optional string key = 1;
-    // string module_name = 2;
-    // If present, the deployment is not yet canonical as it is currently part of a changeset.
-    // optional string changeset = 3;
+// ChangesetFailed is sent when a changeset fails.
+message ChangesetFailedNotification {
+  string key = 1;
+  string error = 2;
+}
 
-    optional string changeset = 1;
-    optional ftl.schema.v1.Module schema = 2;
-  }
+// ChangesetDrained is sent when a changeset has been drained.
+message ChangesetDrainedNotification {
+  string key = 1;
+}
 
-  message DeploymentUpdated {
-    // Will not be set for builtin modules.
-    // optional string key = 1;
-    // string module_name = 2;
-    // If present, the deployment is not yet canonical as it is currently part of a changeset.
-    optional string changeset = 1;
+// ChangesetFinalized is sent when a changeset has been finalized.
+message ChangesetFinalizedNotification {
+  string key = 1;
+}
 
-    optional ftl.schema.v1.Module schema = 2;
-  }
+// CanonicalDeploymentNotification is used to notify the initial schema state
+message CanonicalDeploymentNotification {
```

Comment:
Nit: Sounds oddly singular to me. `CanonicalSchemaNotification`? (I know naming isnt what the point right now...)

Reply:
Actually thats a good point, previously the 'more' flag was used to differentiate, this should just send a single instance of the full schema.